bmc-log-collector/log-collector.go

```diff
@@ -0,0 +1,121 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"log/slog"
+	"net/http"
+	"strconv"
+	"time"
+)
+
+type SystemEventLog struct {
+	Od_Id             string   `json:"@odata.id"`
+	Od_Type           string   `json:"@odata.type"`
+	Create            string   `json:"Created"`
+	Description       string   `json:"Description"`
+	EntryCode         string   `json:"EntryCode"`
+	EntryType         string   `json:"EntryType"`
+	GeneratorId       string   `json:"GeneratorId"`
+	Id                string   `json:"Id"`
+	Message           string   `json:"Message"`
+	MessageArgs       []string `json:"MessageArgs"`
+	OdCnt_MessageArgs int      `json:"MessageArgs@odata.count"`
+	MessageId         string   `json:"MessageId"`
+	Name              string   `json:"Name"`
+	SensorNumber      int      `json:"SensorNumber"`
+	SensorType        string   `json:"SensorType"`
+	Severity          string   `json:"Severity"`
+	Serial            string
+	NodeIP            string
+	BmcIP             string
+}
+
+type RedfishJsonSchema struct {
+	Name        string           `json:"Name"`
+	Count       int              `json:"Members@odata.count"`
+	Context     string           `json:"@odata.context"`
+	Id          string           `json:"@odata.id"`
+	Type        string           `json:"@odata.type"`
+	Description string           `json:"Descriptionta"`
+	Sel         []SystemEventLog `json:"Members"`
+}
+
+// SEL(System Event Log) Collector
+type selCollector struct {
+	machinesListDir string       // Directory of the machines list
+	rfUriSel        string       // SEL path of Redfish API address
+	ptrDir          string       // Pointer store
+	testOutputDir   string       // Test output directory
+	username        string       // iDRAC username
+	password        string       // iDRAC password
+	httpClient      *http.Client // to reuse HTTP transport
+}
+
+func (c *selCollector) selCollectorWorker(ctx context.Context, m Machine, logWriter bmcLogWriter) {
+	bmcUrl := "https://" + m.BmcIP + c.rfUriSel
+	byteJSON, err := requestToBmc(ctx, c.username, c.password, c.httpClient, bmcUrl)
+
+	if err != nil {
+		// When canceled by context, the pointer files is never updated because the return is made here.
+		slog.Error("requestToBmc()", "err", err, "url", c.rfUriSel)
```

Comment:
If errors keep occurring, can the user notice them?

Reply:
avoid logging of a same error